tests/unit/completion/test_sql.py

```diff
@@ -0,0 +1,293 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2016 Ryan Roden-Corrent (rcorre) <ryan@rcorre.net>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Tests for the base sql completion model."""
+
+import pytest
+from PyQt5.QtCore import Qt
+
+from qutebrowser.completion.models import sql
+
+
+def _make_model(data):
+    """Create a completion model populated with the given data.
+
+    Args:
+        data: A list of form
+              [
+                  (cat, [(name, desc, misc), (name, desc, misc), ...]),
+                  (cat, [(name, desc, misc), (name, desc, misc), ...]),
+                  ...
+              ]
+    Returns: (model, categories)
+    """
+    model = sql.SqlCompletionModel()
+    categories = []
+    for title, children in data:
+        categories.append(_add_category(model, title, children))
+    return model, categories
+
+
+def _add_category(model, name, items, **kwargs):
+    """Add a new category to the model with children.
+
+    Args:
+        model: The model to add the category to.
+        name: The title of the category.
+        items: A list of tuples containing column data for each entry.
+        kwargs: Keyword arguments passed through to model.new_category.
+    """
+    cat = model.new_category(name, **kwargs)
+    for item in items:
+        cat.new_item(*item)
+    return cat
+
+
+def _check_model(model, expected):
+    """Check that a model contains the expected items in any order.
+
+    Args:
+        expected: A list of form
+                  [
+                      (cat, [(name, desc, misc), (name, desc, misc), ...]),
+                      (cat, [(name, desc, misc), (name, desc, misc), ...]),
+                      ...
+                  ]
+    """
+    assert model.rowCount() == len(expected)
+    for i, (expected_title, expected_items) in enumerate(expected):
+        catidx = model.index(i, 0)
+        assert model.data(catidx) == expected_title
+        assert model.rowCount(catidx) == len(expected_items)
+        for j, (name, desc, misc) in enumerate(expected_items):
+            assert model.data(model.index(j, 0, catidx)) == name
+            assert model.data(model.index(j, 1, catidx)) == desc
+            assert model.data(model.index(j, 2, catidx)) == misc
+
+
+@pytest.yield_fixture(autouse=True)
```

Comment:
`@pytest.yield_fixture` has been [merged](https://github.com/pytest-dev/pytest/issues/1461) into `@pytest.fixture` with pytest 3.0, so you can simply use that.